python/mxnet/module/module.py

```diff
@@ -171,6 +174,77 @@ def save_checkpoint(self, prefix, epoch, save_optimizer_states=False):
             self.save_optimizer_states(state_name)
             logging.info('Saved optimizer state to \"%s\"', state_name)
 
+    def export_serving(self, prefix, epoch, signature, save_optimizer_states=False,
+                       use_synset=False):
+        """Export current progress to a zip file used by MXNet model serving.
+
+        Parameters
+        ----------
+        prefix : str
+            The file prefix to export to.
+        epoch : int
+            The current epoch number.
+        signature : dict
+            A dictionary containing model input and output information.
+            Data names or data shapes of inputs and outputs can be automatically
+            collected from module. User needs to specify inputs and outputs MIME
+            type for http request. Currently only 'image/jpeg' and 'application/json'
+            are supported. All inputs should have the same type. This also applies for
+            outputs.
+            An example signature would be:
+                signature = { "input_type": "image/jpeg", "output_type": "application/json" }
+        save_optimizer_states : bool
+            Whether to save optimizer states to continue training.
+        use_synset : bool
```

Comment:
Classification class labels can be stored in files with varying names, not only synset.txt.
Why do we call it "use_synset" and hard code the class labels?
I suggest we have an optional argument class_labels : string instead

Reply:
This may be a good chance to refactor this interface. Class label is only for classification task, but this API is a general one for all DL models. For example, users may want to put vocabulary dictionary file into exported model as well. Sometimes they may put multiple files into exported model. My suggestion is we can change this argument to 'util_file', which is a list of files users want exported model to contain. In this way, users can use these files in MMS, though currently only classification label will be automatically loaded.